code/modules/mob/living/carbon/human/update_icons.dm

```diff
@@ -767,22 +765,22 @@ var/global/list/damage_icon_parts = list()
 		if(back.item_state_slots && back.item_state_slots[slot_back_str])
 			overlay_state = back.item_state_slots[slot_back_str]
 
-
 		//apply color
 		var/image/standing = image(icon = overlay_icon, icon_state = overlay_state)
 		standing.color = back.color
 
-		//create the image
+		if(back)//rig's sprite by itself, then moduls
+			if(istype(back, /obj/item/weapon/rig))
```

Comment:
This is bad. In future i'm planning to add rigs that are worn on slots other than the back. This is why rigs have a var to indicate which slot they should be worn in.

Specific case code like this in update inv procs is also generally bad.
This should call a generic proc on the worn item, and that item should make its own overlays as needed

Reply:
There was the same code before, if you know how to evade checks  like this, and make this proc clearer - tell me, i'm already bored with all this trashy and old code which i wanted to "refactor" but discarded all changes becouse firstly i want this PR.

Reply:
Call a generic function

if (back)
back.apply_overlays(standing)

This icon updating code is generic, it needs to not have special cases like this in it

Reply:
In this case i rerally don't like to make the only one function for one-time use, as for me it is clearer to take one string in this place. Btw `if(back)` is really unnecessary here, will fix soon.